### PR TITLE
[FLINK-36128] Promote LENIENT as the default schema change behavior

### PR DESCRIPTION
**Pull request by yuxiqian**
This closes FLINK-36128.

Currently, default schema evolution mode "EVOLVE" could not handle exceptions, and might not be able to restore from existing state correctly after failover. Before we can "trigger checkpoints on demand" which is not possible prior to in Flink 1.19, making "LENIENT" a default option might be more suitable.

This PR also contains a hotfix patch for PrePartitionOperator broadcasting failure.

### CODE REVIEW

File: flink-cdc-cli/src/main/java/org/apache/flink/cdc/cli/parser/YamlPipelineDefinitionParser.java [hunk 1]
```diff
@@ -172,6 +183,23 @@ private SinkDef toSinkDef(JsonNode sinkNode) {
...
+            // default.
+            Arrays.stream(SchemaChangeEventTypeFamily.ALL)
+                    .map(SchemaChangeEventType::getTag)
+                    .forEach(includedSETypes::add);
+        }
+
+        if (excludedSETypes.isEmpty()
+                && SchemaChangeBehavior.LENIENT.equals(schemaChangeBehavior)) {
+            // In lenient mode, we exclude DROP_TABLE and TRUNCATE_TABLE by default. This could be
+            // overridden by manually specifying excluded types.
```
Comment (leonardBang) [inline flink-cdc-cli/src/main/java/org/apache/flink/cdc/cli/parser/YamlPipelineDefinitionParser.java:197]: minor: we should add schema behavior  document ASAP

**Review by leonardBang — APPROVED** [review]
Thanks @yuxiqian for the fix, + 1 for the point we have to enable lenient mode as default before flink 1.19, LGTM

File: flink-cdc-cli/src/main/java/org/apache/flink/cdc/cli/parser/YamlPipelineDefinitionParser.java (hunk 1 above)
Comment (yuxiqian) [reply flink-cdc-cli/src/main/java/org/apache/flink/cdc/cli/parser/YamlPipelineDefinitionParser.java:197]: Tracked with FLINK-36151.

**Review by leonardBang — APPROVED** [review]
CI passed, merging

**Pull request merged**